src/main/frontend/components/search.cljs

```diff
@@ -83,7 +85,137 @@
 
 (defonce search-timeout (atom nil))
 
-(rum/defc ^:large-vars/cleanup-todo search-auto-complete
+(defn- search-on-chosen-open-link
+  [repo search-q {:keys [data type alias]}]
+  (search-handler/add-search-to-recent! repo search-q)
+  (search-handler/clear-search!)
+  (case type
+    :block
+    ;; Open the first link in a block's content
+    (let [block-uuid (uuid (:block/uuid data))
+          block (:block/content (db/entity [:block/uuid block-uuid]))
+          link (re-find editor-handler/url-regex block)]
```

Comment:
I thought about using `editor-handler/extract-nearest-link-from-text` but I didn't really need of any it as I don't care about cursor position or any of the other link types